src/libraries/System.IO.Compression/src/System/IO/Compression/ZipHelper.cs

```diff
@@ -103,91 +103,85 @@ internal static uint DateTimeToDosTime(DateTime dateTime)
         // assumes maxBytesToRead is positive, ensures to not read beyond the provided max number of bytes,
         // if the signature is found then returns true and positions stream at first byte of signature
         // if the signature is not found, returns false
-        internal static bool SeekBackwardsToSignature(Stream stream, uint signatureToFind, int maxBytesToRead)
+        internal static bool SeekBackwardsToSignature(Stream stream, ReadOnlySpan<byte> signatureToFind, int maxBytesToRead)
         {
-            Debug.Assert(signatureToFind != 0);
+            Debug.Assert(signatureToFind.Length != 0);
             Debug.Assert(maxBytesToRead > 0);
 
             int bufferPointer = 0;
-            uint currentSignature = 0;
-            byte[] buffer = new byte[BackwardsSeekingBufferSize];
+            byte[] buffer = ArrayPool<byte>.Shared.Rent(BackwardsSeekingBufferSize);
+            Span<byte> bufferSpan = buffer.AsSpan(0, BackwardsSeekingBufferSize);
 
-            bool outOfBytes = false;
-            bool signatureFound = false;
-
-            int bytesRead = 0;
-            while (!signatureFound && !outOfBytes && bytesRead <= maxBytesToRead)
+            try
             {
-                outOfBytes = SeekBackwardsAndRead(stream, buffer, out bufferPointer);
+                bool outOfBytes = false;
+                bool signatureFound = false;
 
-                Debug.Assert(bufferPointer < buffer.Length);
+                int totalBytesRead = 0;
+                int duplicateBytesRead = 0;
 
-                while (bufferPointer >= 0 && !signatureFound)
+                while (!signatureFound && !outOfBytes && totalBytesRead <= maxBytesToRead)
                 {
-                    currentSignature = (currentSignature << 8) | ((uint)buffer[bufferPointer]);
```

Comment:
Glad we're getting rid of this line. Not a fan of bit shifting operations when there are easier-to-read alternatives.